### PR TITLE
Update fused kernels and call _safe_softmax from SDPA

### PR DESCRIPTION
**Pull request by drisspg**
Summary:
X-link: https://github.com/pytorch/pytorch/pull/131863

cc ezyang gchanan jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 Valentine233

imported-using-ghimport

Reviewed By: Chillee

Differential Revision: D61418679

Pulled By: drisspg

### CODE REVIEW

**Comment by shoumikhin** [comment]
@YifanShenSZ ptal when you have a chance!

File: backends/apple/coreml/test/test_coreml_partitioner.py [hunk 1]
```diff
@@ -68,15 +68,23 @@ def test_vit_skip_conv(self):
...
 
+        conv_block = ["aten.convolution.default", "executorch_call_delegate"]
+        safe_softmax_block = [
+            "getitem",
+            "getitem",
+            "getitem",
+            "getitem",
+            "aten.any.dim",
+            "executorch_call_delegate",
+        ]
```
Comment (YifanShenSZ) [inline backends/apple/coreml/test/test_coreml_partitioner.py:79]: In the image @shoumikhin showed me
image
I thought each safe softmax block should be
```
"executorch_call_delegate",
"getitem",
"aten.any.dim",
"executorch_call_delegate",
"getitem",
```
?

Comment (drisspg) [reply backends/apple/coreml/test/test_coreml_partitioner.py:79]: So the script that generated this https://gist.github.com/drisspg/b4087fc8aad510926c2cbe24e89f8e3b since I was having trouble building the actual PR with executorch: https://github.com/pytorch/pytorch/pull/131863

Comment (YifanShenSZ) [reply backends/apple/coreml/test/test_coreml_partitioner.py:79]: Make sense 👍 then as long as CI green I believe this is good

**Review by YifanShenSZ — APPROVED** [review]
Awesome safe softmax 🎊 & thanks for the subsequent fix in executorch 🙏

**Comment by drisspg** [comment]
@shoumikhin not totally sure what to do about how red executorch trunk is on this one

**Review by shoumikhin — APPROVED** [review]
